### ISSUE TITLE
linter: add support "files" in config

### ISSUE THREAD

**Issue by pumano** [oxc-project/oxc#7096]
### What version of Oxlint are you using?

0.11.0

### What does your `.oxlint.json` config file look like?

Proposed config:

```json
{
  "$schema": "./node_modules/oxlint/configuration_schema.json",
  "plugins": ["import", "unicorn"],
  "files": ["**/*.ts", "**/*.tsx"],
  "rules": {
    "import/no-commonjs": "error",
    "import/no-cycle": "error",
    "import/no-duplicates": "error",
    "import/no-self-import": "error"
  }
}
```

### What happened?

"files" should be supported, because it target some rules for specific file types. In my case i use import/no-commonjs rule against only `*.ts` files and skip `*.js` files

**Comment by pumano** [comment]
looks like `"files"` covered somehow only in upcoming `"overrides"` section: https://github.com/oxc-project/oxc/pull/6974/
but not in root too.

**Comment by DonIsaac** [comment]
Should be trivial to add once overrides PR is in. 

Apologies, that one is blocked by me- I still need to perform more intense testing

**Comment by pumano** [comment]
@Sysix if it's flat, I really have no problem, is can be postponed, because I use "override" with "files" if I need specific file types

**Comment by Boshen** [comment]
Has a workaround: "overrides".
